Error build in OpenBSD

SoftEther version: 5.01 Build 9663
Component: Server
Operating system: OpenBSD 6.4
Architecture: 64 bit

[ 65%] Building C object src/Mayaqua/CMakeFiles/mayaqua.dir/Encrypt.c.o
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:427:33: warning: incompatible pointer types passing
      'const struct evp_md_st *' to parameter of type 'const EVP_MD *' (aka 'const struct env_md_st *')
      [-Wincompatible-pointer-types]
                if (EVP_DigestInit_ex(m->Ctx, m->Md, NULL) == false)
                                              ^~~~~
/usr/include/openssl/evp.h:556:54: note: passing argument to parameter 'type' here
int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type, ENGINE *impl);
                                                     ^
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:666:35: error: use of undeclared identifier 'EVP_CTRL_AEAD_SET_TAG'
                if (EVP_CIPHER_CTX_ctrl(c->Ctx, EVP_CTRL_AEAD_SET_TAG, tag_size, tag) == false)
                                                ^
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:696:35: error: use of undeclared identifier 'EVP_CTRL_AEAD_GET_TAG'
                if (EVP_CIPHER_CTX_ctrl(c->Ctx, EVP_CTRL_AEAD_GET_TAG, tag_size, tag) == false)
                                                ^
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:4496:8: warning: incompatible pointer types assigning to
      'const struct evp_md_st *' from 'const EVP_MD *' (aka 'const struct env_md_st *')
      [-Wincompatible-pointer-types]
        m->Md = md;
              ^ ~~
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:4505:27: warning: passing 'const void *' to parameter of type 'void *'
      discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        len = MdProcess(m, dest, src, src_size);
                                 ^~~
/home/SoftEtherVPN/src/Mayaqua/Encrypt.c:456:42: note: passing argument to parameter 'src' here
UINT MdProcess(MD *md, void *dest, void *src, UINT size)
                                         ^
3 warnings and 2 errors generated.
*** Error 1 in . (src/Mayaqua/CMakeFiles/mayaqua.dir/build.make:87 'src/Mayaqua/CMakeFiles/mayaqua.dir/Encrypt.c.o': cd /home/SoftEtherVPN/t...)
*** Error 1 in . (CMakeFiles/Makefile2:219 'src/Mayaqua/CMakeFiles/mayaqua.dir/all')
*** Error 1 in /home/SoftEtherVPN/tmp (Makefile:152 'all')

--- a/src/Mayaqua/Encrypt.h
+++ b/src/Mayaqua/Encrypt.h
@@ -239,13 +239,19 @@ void RAND_Free_For_SoftEther();
 
 // OpenSSL <1.1 Shims
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
-#	define EVP_CTRL_AEAD_GET_TAG EVP_CTRL_GCM_GET_TAG
-#	define EVP_CTRL_AEAD_SET_TAG EVP_CTRL_GCM_SET_TAG
 #	define EVP_PKEY_get0_RSA(obj) ((obj)->pkey.rsa)
 #	define EVP_PKEY_base_id(pkey) ((pkey)->type)
 #	define X509_get0_notBefore(x509) ((x509)->cert_info->validity->notBefore)
 #	define X509_get0_notAfter(x509) ((x509)->cert_info->validity->notAfter)
 #	define X509_get_serialNumber(x509) ((x509)->cert_info->serialNumber)
+#endif
+
+#ifndef EVP_CTRL_AEAD_GET_TAG
+#	define EVP_CTRL_AEAD_GET_TAG EVP_CTRL_GCM_GET_TAG
+#endif
+
+#ifndef EVP_CTRL_AEAD_SET_TAG
+#	define EVP_CTRL_AEAD_SET_TAG EVP_CTRL_GCM_SET_TAG
 #endif
 
 // Crypt context

